_posts/2024-05-22-developing-openwrt-in-qemu.adoc

```diff
@@ -0,0 +1,93 @@
+---
+title: Developing OpenWRT in QEMU with Eclipse Che
+layout: post
+author: Oleksii Orel
+description: >-
+  Developing OpenWRT helloworld package in QEMU machine emulator and virtualizer with Eclipse Che.
+categories: []
+keywords: ['openwrt', 'quemu', 'gdb', 'developing ', 'workspace']
+slug: /@olexii.orel/openwrt-helloworld-package
+---
+
+https://openwrt.org/[OpenWrt] is a highly extensible GNU/Linux distribution for embedded devices (typically wireless routers). OpenWrt is built from the ground up to be a full-featured, easily modifiable operating system for embedded devices. You can have all the features you need with no bloat, powered by a modern Linux kernel.
+
+We can create and develop https://openwrt.org/docs/guide-developer/creating-a-cmake-package-in-openwrt[OpenWRT CMake packages] with Eclipse Che.
+
+In this blog post, we will review developing OpenWRT in https://www.qemu.org/[QEMU] with Eclipse Che in 4 parts:
+
+<<part1>>
+
+<<part2>>
+
+<<part3>>
+
+<<part4>>
+
+
+NOTE: In this blog post, we will use https://github.com/che-incubator/openwrt-helloworld-package.git[OpenWrt HelloWorld package].
+
+
+First of all, we need to create a new workspace from the repo https://github.com/che-incubator/openwrt-helloworld-package.git[openwrt-helloworld-package] with Eclipse Che. The easiest way to do it is to use https://eclipse.dev/che/docs/stable/hosted-che/hosted-che/[Eclipse Che hosted by Red Hat] and create a workspace by navigating to the following URL - https://workspaces.openshift.com#https://github.com/che-incubator/openwrt-helloworld-package/blob/main/devfile.yaml
```

Comment:
I get an error when I try this link:
![Screenshot from 2024-05-21 11-56-08](https://github.com/eclipse-che/blog/assets/83611742/aafce862-aaaf-4ad6-81fe-c5264071881a)

I have no problems if I use this link however:
```
https://workspaces.openshift.com#https://github.com/che-incubator/openwrt-helloworld-package/
```

Reply:
Thanks. I have fixed it.